lib/completions/llm.rb

```diff
@@ -174,6 +175,7 @@ def generate(
 
         model_params.keys.each { |key| model_params.delete(key) if model_params[key].nil? }
 
+        gateway = @gateway || gateway_klass.new(model_name, dialect_klass.tokenizer)
```

Comment:
Feels like it's easy to get in a situation where `@gateway` is nil and `gateway_klass` is nil too

Reply:
yeah I just added that for testing, I think the testing pattern is off but that is a bigger refactor 

Thanks heaps